src/main/java/nextstep/subway/applicaion/LineService.java

```diff
@@ -0,0 +1,72 @@
+package nextstep.subway.applicaion;
+
+import nextstep.subway.applicaion.dto.LineRequest;
+import nextstep.subway.applicaion.dto.LineResponse;
+import nextstep.subway.applicaion.dto.LineUpdateRequest;
+import nextstep.subway.domain.Line;
+import nextstep.subway.domain.LineRepository;
+import nextstep.subway.domain.Station;
+import nextstep.subway.domain.StationRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+@Service
+@Transactional(readOnly = true)
+public class LineService {
+
+    private final LineRepository lineRepository;
+    private final StationRepository stationRepository;
+
+    public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+        this.lineRepository = lineRepository;
+        this.stationRepository = stationRepository;
+    }
+
+    @Transactional
+    public LineResponse saveLine(LineRequest lineRequest) {
+        validationStations(lineRequest.getUpStationId(), lineRequest.getDownStationId());
+        Station upStation = findStation(lineRequest.getUpStationId());
+        Station downStation = findStation(lineRequest.getDownStationId());
+
+        final Line line = new Line(lineRequest.getName(), lineRequest.getColor(), lineRequest.getDistance(), upStation, downStation);
```

Comment:
여기서 하나 궁금한게 있습니다! 😊

평소에 다른 개발자의 코드를 보는걸 좋아해서, 요번에도 한번 다 구현하고, 다른 교육생분들은 어떻게 구현하는지 봤는데,

```suggestion
        final Line line = LineRequest.toDomain();
```

이렇게도 쓰시더라구요!  저도 한번 써보려고 생각이 들었는데, LineRquest에 toDomain이라는 메서드가 존재하면, 더 복잡해지는 느낌이 들더라구요. 리뷰어님 의견이 궁금해요! 😅

Reply:
해당 부분은 의견이 다른 분들을 많이 보았는 데, 제 개인적인 생각을 말씀 드리겠습니다.
도메인 객체의 생성에 대한 책임을 어느 객체가 가질 것인가에 대한 이야기로 생각합니다.
일단 제가 알고 있는 방법들은 크게 세가지 정도인데요

1. DTO
    객체 생성에 대한 책임을 DTO 로 준 경우인데요. 조졸두 님 책에 수록되어 있기도 해서 많이 쓰시는 것 같습니다.
    장점은 자주 변경되는 DTO 객체에서 변환을 담당하여 변경 시 수용이 편하다는 점이 있다고 생각합니다.(entity 를 만드는 방법 등을 쉽게 변경할 수 있음)
    단점으로는 요청이 항상 특정 Entity 에 강하게 결합될 가능성이 있을 것 같습니다.
2. Service
    서비스 레이어에서만 Entity 를 감싸고 있기 때문에, 안정적이라고 생각됩니다.
    다만, 객체 생성 로직이 복잡해질수록 서비스 레이어가 지저분해질 가능성이 있다고 생각합니다.
3. Factory 
    객체 생성을 위해 따로 Factory 를 구축하는 방법인데, 직접 구현할 수도 있고, `ModelMapper` 나 `MapStruct` 등을 사용할 수 있습니다.
     간단하게 객체를 생성할 때는 과하다고 느껴질 수 있습니다.

다른 사람이 생각하는 장/단점 또한 많이 찾아보시고 고민해보시면 더 좋은 코드가 나올 것이라고 생각합니다 😄

Reply:
크.. 좋은 의견 감사합니다. 요즘 코드를 코드를 짜거나, 코드를 볼 때, 그리고 코드 리뷰를 할 때, 쓰는 사람이 각기 다른 방법을 사용하더라고요. 하지만, 이러한 방법이 실무에서 각각 놀고 있으면 안 되고 팀 내 컨벤션을 작성하여 같이 같은 방식으로 해 나아가야 된다는 생각이 요즘 들었어요. 이러한 생각을 가지고, 주위에 있는 분들의 의견, 그리고 리뷰어님의 의견을 다 듣고 싶고, 저 또한 어떠한 방법이 좋은지에 선택하고 싶었어요.

소중한 의견 너무 감사합니다. 리뷰어님, 리뷰어님 덕분에 요번에도 한층 더 성장한 느낌입니다! 😊